exercicios/para-casa/Sayonara/SAYONARA.sistema_de_alunas.py

```diff
@@ -0,0 +1,152 @@
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 5).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+            
+        return codigo_opcao
+    
+def incluir_nova_aluna():
+    print("Informe os dados ")
+    nome= input("Informe o nome e o sobrenome :")
+    turma = input("informe a turma :" )
+    notas = pedir_notas()
+    nota_participaçao=float(input("participação da aluna: "))
+    presença= chamada() 
+    dataset[(nome.split()[0] , nome.split()[1])] = {
+        "Turma": turma,
+         "Notas": notas,
+         "Presença": presença,
+         "Paticipaçao": nota_participaçao
+    }
+    print("Aluna cadastrada com sucesso ")
+    pass
+    #TODO - Implentar a função
+def pedir_notas():
+   #quantidade_notas=input("informa a quantidade das notas:")
+   notas=[]
+   for i in range(3):
+         nota=float(input(f"Digite a {i+1}° nota:" ))
+         notas.append(nota)
+
+
+   return notas 
+ 
+def chamada(): 
+    aulas=[] 
+    for  i in range (5): 
+     aula=int(input(f" Digite 1 se presente ou 0 se ausente, na aula :{i+1}°: presença:"))  
+    aulas.append(aula)
+    return aulas 
+
+    
+def consultar_lista_alunas(dataset):
+     print("Lista alunas ")
+     for aluna_tuple in dataset:
+        nome_c = aluna_tuple[ 0 ]  + " " + aluna_tuple[1]
+        nome =nome_c.strip('(').strip()
+        print (nome)
+
+     pass
+    #TODO - Implentar a função
+    
+def consultar_faltas_aluna():
+     print("informe o nome da aluna")
+     nome_completo=input("Nome completo")
+     nome_aluna,sobrenome_aluna=nome_completo.split()
+     chave_aluna=(nome_aluna,sobrenome_aluna)
+
+     if chave_aluna in dataset:
+      presencas=dataset[chave_aluna]["Presença"]
+      numero_faltas=presencas.count(False)
+      print(f" Número de faltas de {nome_aluna}  {sobrenome_aluna} :{numero_faltas}")
+     
+
+            
+        
+     return
+
+     pass
+    #TODO - Implentar a função
+    
+def consultar_notas_aluna():
+    print("Informe o nome da aluna")
+    nome_completo=input("Nome completo: ")
+    nome_aluna,sobrenome_aluna=nome_completo.split()
+    chave_aluna=(nome_aluna,sobrenome_aluna)
+    if chave_aluna in dataset:
+      nota=dataset[chave_aluna]["Notas"]
+
+
+
+
+
+      print(f" Notas de {nome_completo}:{nota}" )
+
+ 
+
+
+
+
+   
+    return
+    pass
+    #TODO - Implentar a função
+    
+def consultar_status_aprovacao():
+     
+     print("Informe o nome da aluna")
+     nome_completo=input("Nome Completo")
+     nome_aluna,sobrenome_aluna=nome_completo.split()
+     chave_aluna=(nome_aluna,sobrenome_aluna)
+     if chave_aluna in dataset:
+      notas=dataset[chave_aluna]["Notas"]
+      nota_participaçao=dataset[chave_aluna]["Participação"]
+      faltas=dataset[chave_aluna]["Presença"]
+      faltas_contagem=faltas.count(False)
+      media=sum(notas)/len(notas)
+      if media>=6 and nota_participaçao>=6 and faltas_contagem<=2:
```

Comment:
Neste ponto, era esperado que o cálculo de faltas fosse realizado pelo percentual e não pela quantidade. No requisito funcional é esperado que o cálculo fosse de 80% de presença. Futuramente, com o aumento do dataset, se incluirmos mais dados de presença, esse valor de 2 faltas ficará errado.
Entre operadores, procure adicionar espaços para ajudar na legibilidade do código.

**Sugestão:**
```suggestion
     percentual_faltas = dataset[chave_aluna]["Presença"].count(False) / len(dataset[chave_aluna]["Presença"])
      if media >= 6 and nota_participaçao >= 6 and percentual_faltas <= 0.2:
```